sql/sys_vars.h

```diff
@@ -1107,6 +1107,20 @@ class Sys_var_charptr : public sys_var {
 
   bool global_update(THD *thd, set_var *var) override;
 
+  bool global_update_value(const char *const ptr) {
+    char *new_val;
+    if (ptr) {
+      new_val = my_strdup(key_memory_Sys_var_charptr_value, ptr, MYF(MY_WME));
+      if (!new_val) return true;
+    } else {
+      new_val = nullptr;
+    }
+    if (flags & ALLOCATED) my_free(global_var(char *));
+    flags |= ALLOCATED;
+    global_var(char *) = new_val;
+    return false;
+  }
+
```

Comment:
Did we have this in 5.7 PR? Can you please point me to it?

Reply:
We don't have it for 5.7 (probably missed it while moving code to 5.7) but it looks like ALLOCATED flag should be handled for 5.7 as well (even though I don't see ALLOCATED to be actually used for any sysvar).

Reply:
if it is not used, and it can work (check other variables) without setting ALLOCATED flag, better to remove the entire func in 8.0 too

Reply:
@satya-bodapati, do you think it would be wise to remove this ALLOCATED flag at all (in vase it is really not used anywhere)?

Reply:
> better to remove the entire func in 8.0 too

Done

Reply:
btw, I see some other places where it is used. not sure if we can remove it.